hearth/lib/hearth/http/client.rb

```diff
@@ -47,22 +47,21 @@ class Client
       #   optionally other keyword args similar to {Addrinfo#getaddrinfo}'s
       #   positional parameters.
       def initialize(options = {})
-        @http_wire_trace = options[:http_wire_trace]
-        @logger = options[:logger]
-        @http_proxy = URI(options[:http_proxy]) if options[:http_proxy]
-        @ssl_verify_peer = options[:ssl_verify_peer]
-        @ssl_ca_bundle = options[:ssl_ca_bundle]
-        @ssl_ca_directory = options[:ssl_ca_directory]
-        @ssl_ca_store = options[:ssl_ca_store]
+        @debug_output = options[:debug_output]
+        @proxy = URI(options[:proxy]) if options[:proxy]
+        @verify_peer = options[:verify_peer]
+        @ca_file = options[:ca_file]
+        @ca_path = options[:ca_path]
+        @cert_store = options[:cert_store]
         @host_resolver = options[:host_resolver]
       end
 
       # @param [Request] request
       # @param [Response] response
       # @return [Response]
-      def transmit(request:, response:)
+      def transmit(request:, response:, **options)
         http = create_http(request.uri)
-        http.set_debug_output(@logger) if @http_wire_trace
+        http.set_debug_output(options[:logger]) if @debug_output
```

Comment:
What happens here if logger is nil?

Reply:
It shouldn't be, but I believe the method succeeds but no logging will occur.